Dockerfile

```diff
@@ -0,0 +1,31 @@
+# extend node alpine base image
+FROM node:8.4.0-alpine
+
+ENV RENOVATE_VERSION 9.45.10
+
+MAINTAINER Rhys Arkins <rhys@arkins.net>
+LABEL NAME renovate
+LABEL VERSION $RENOVATE_VERSION
+
+# globally install the specified version of renovate
+RUN set -x \
+  && npm install -g renovate@$RENOVATE_VERSION
+
+# inject the custom entrypoint
+# ensure that it is executable by user 1000
+# user 1000 is already setup for usage in node base images
+COPY entrypoint.sh /entrypoint.sh
+RUN set -x \
+  && chown 1000:0 /entrypoint.sh \
+  && chmod u+x /entrypoint.sh
+
+# switch to user 1000
+# this prevents several problems in specialized environments
+USER 1000
```

Comment:
Thank you for switching to user `1000` (which can also be specified as `node`). It prevents the subsequent commands from being executed as root, which can be a security issue.

Docker's _Security_ guide has a [Conclusions](https://docs.docker.com/engine/security/security/#conclusions) section that recommends, for added security, configuring a non-root user to run commands.

I wonder if it could even be called before running `npm install --global` so that `renovate` is installed globally for the user, and not executed as `root`.